client/http/request_info.go

```diff
@@ -88,11 +89,14 @@ type requestInfo struct {
 	res         any
 	respHandler respHandleFunc
 	bo          *retry.Backoffer
+	timeout     time.Duration
 }
 
 // newRequestInfo creates a new request info.
 func newRequestInfo() *requestInfo {
-	return &requestInfo{}
+	return &requestInfo{
+		timeout: defaultTimeout,
```

Comment:
What do you think about setting a default timeout for the client and passing it through? Leave the`requestInfo` always empty to customize.

Reply:
+1

Reply:
Do you mean that it is better to set default timeout for `http.Client`?
I can have a try, but we should replace `innter client` to do it.
The reason why I made this change is because:
1. I just want to initialize an `http.Client`
2. The timeout value of `http.Client` used in TiDB is 0
3. The timeout value of `http.Client` used in client-go is 30s